### PR TITLE
refactor(table): remove Table impl for system

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

* Remove `impl Table for SystemCatalogTable`
* Move down methods like `register_table` to `SystemCatalogTable`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/2065

### CODE REVIEW

File: src/catalog/src/system.rs [hunk 1]
```diff
@@ -21,24 +20,23 @@ use common_catalog::consts::{
     SYSTEM_CATALOG_NAME, SYSTEM_CATALOG_TABLE_ID, SYSTEM_CATALOG_TABLE_NAME,
 };
 use common_recordbatch::SendableRecordBatchStream;
-use common_telemetry::debug;
+use common_telemetry::{debug, logging};
```
Comment (waynexia) [inline src/catalog/src/system.rs:23]: you can import `warn` directly

File: src/catalog/src/system.rs [hunk 2]
```diff
@@ -126,6 +88,54 @@ impl SystemCatalogTable {
...
+    pub(crate) async fn deregister_table(
+        &self,
+        request: &DeregisterTableRequest,
+        table_id: TableId,
+    ) -> Result<()> {
+        let deletion_request = build_table_deletion_request(request, table_id);
+        self.0
+        .insert(deletion_request)
+        .await
+        .map(|x| {
```
Comment (waynexia) [inline src/catalog/src/system.rs:116]: Looks like L114-L116 should have one more tab...?

Comment (zhongzc) [reply src/catalog/src/system.rs]: Ah.. what happened to formatter

**Review by waynexia — APPROVED** [review]
LGTM

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**